generators/templates.go

```diff
@@ -419,14 +429,18 @@ type {{ initialCap .Name }} struct {
 
 {{ if .LinkSets | len }}
 type {{ initialCap .Name }}LinkSets struct {
-  {{ range .LinkSets }} {{ initialCap .Name }}	map[string]modeldb.Link		` + "`" + `json:"{{ .Name }},omitempty"` + "`" + `
+  {{ range .LinkSets }}
+  {{ .Name }}Mutex sync.Mutex
+  {{ initialCap .Name }}	map[string]modeldb.Link		` + "`" + `json:"{{ .Name }},omitempty"` + "`" + `
   {{end}}
 }
 {{ end }}
 
 {{ if .Links | len }}
 type {{ initialCap .Name }}Links struct {
-  {{ range .Links }} {{ initialCap .Name }} modeldb.Link	` + "`" + `json:"{{ .Name }},omitempty"` + "`" + `
+  {{ range .Links }} 
+  {{ .Name }}Mutex sync.Mutex
```

Comment:
I think we probably don't need this lock. If we are thinking about protecting the whole object it should be at a higher level.